### ISSUE TITLE
wander --version shows "devel" when built from a tag tarball

### ISSUE THREAD

**Issue by classabbyamp** [robinovitch61/wander#102]
**Describe the bug**

When built outside of the git repo, Buildinfo doesn't populate the version info properly, so the version info is wrong.

**To reproduce**
1. download the source code tarball generated by github for a tag (example: https://github.com/robinovitch61/wander/archive/refs/tags/v0.11.1.tar.gz)
2. extract and build
3. `wander --version` returns `wander version devel`

**Expected behavior**

`wander --version` returns `wander version v0.11.1`

**Please complete the following information:**
 - OS: Void Linux
 - Terminal: irrelevant
 - Version: v0.11.1 bilt from source
 - Configuration options used: irrelevant

**Comment by robinovitch61** [comment]
thanks for this @classabbyamp , I'll look into fixing it asap

**Comment by robinovitch61** [comment]
comparing wander with another popular tui, `glow`, this behavior is replicable unfortunately, even when cloning the git repo and building off a tagged commit, as they embed version info using [ldflags in goreleaser](https://github.com/charmbracelet/meta/blob/main/goreleaser-glow.yaml#L34)

```sh
git clone https://github.com/charmbracelet/glow.git
cd glow
git checkout ad21129703d0bd48a9b45227f56d2737623db57b
go build
./glow --version
glow version unknown (built from source)
```

The way I do it in wander is a bit better - if there is a git repo, the version will include the hash:
```
git clone --depth 1 git@github.com:robinovitch61/wander.git
cd wander
go build
./wander --version
wander version rev-09a66a4
```

That's because I'm using the [versioninfo package](https://blog.carlmjohnson.net/post/2023/golang-git-hash-how-to/)

> Fortunately, with the changes brought in with Go 1.18, now there is a special field just for the main module that will include version control information if it’s available...
> One, you can always opt out of including version control information by passing -buildvcs=false to go build. By default, tests don’t include version control information, because it can take some time to include it, and you usually don’t want it.
> Two, the build information will just be a Git revision hash (or equivalent for other version control systems), not a Git tag, because you could have multiple tags for the current commit, but there is only one hash for the current commit, so the hash is less ambiguous.
> Three, on the other hand, when you install a remote Go binary using go install example.com/cmd/sometool@v1.2.3, it won’t have any version control information, since it was downloaded and built from Go modules, and not from a version control system. In those cases, you can instead read the BuildInfo.Main.Version which will have the Git tag that was used to publish as a Go module.

My question for you @classabbyamp is why want to build from source code without cloning the git repo? Do any of these alternatives work for you?
* `go install github.com/robinovitch61/wander@v0.11.1`
* `go install github.com/robinovitch61/wander@latest`
* the git clone method to get the exact commit rather than the latest tag:
```
git clone --depth 1 git@github.com:robinovitch61/wander.git
cd wander
go build
./wander --version
wander version rev-09a66a4
```
* downloading the pre-built linux binary from the releases page directly, e.g. `wander_0.11.1_Linux_*.tar.gz`

**Comment by classabbyamp** [comment]
I am building this as a package for void linux: https://github.com/void-linux/void-packages/blob/master/srcpkgs/wander/template (the script that does the go building is here: https://github.com/void-linux/void-packages/blob/master/common/build-style/go.sh)

by policy, we do not use git repositories or `go install` as sources for our packages, as we like to be able to do things like cache the tarball. the usual way we do version embedding for go packages is by using go ldflags to specify the version. I guess my preferred solution would be to have an ldflag fallback for this, so that it can be specified when buildinfo won't work

**Comment by robinovitch61** [comment]
Gotcha!

> I guess my preferred solution would be to have an ldflag fallback for this, so that it can be specified when buildinfo won't work

by this do you mean that you'll specify your own ldflags to `go build`, something like `go build -ldflags="-X 'main.Version=v0.11.1'"`?

**Comment by classabbyamp** [comment]
correct, that solution would work nicely

**Comment by classabbyamp** [comment]
thanks! we'll pick it up in the next release

**Comment by robinovitch61** [comment]
sweet, released in v0.11.2 with [docs in readme](https://github.com/robinovitch61/wander#manually-specifying-the-wander-version)
